File: Leetcode-Medium/Container-WithMostWater/R.c
int maxArea(int* height, int heightSize) {
    int i = 0;
    int j = heightSize - 1;
    int best = 0;

    while (i < j) {
        int h = height[i] < height[j] ? height[i] : height[j];
        int w = j - i;
        int area = h * w;

        if (area > best) best = area;

        if(height[i]  < height[j]) i++;
        else 
           j--;
    }
    return best;
}

File: Leetcode-Medium/Container-WithMostWater/R.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        best = 0
        i = 0
        j = len(height) - 1
        while i < j:
            h = min(height[i], height[j])
            width = j - i
            best = max(best, width * h)

            if height[i] < height[j]:
                i += 1
            else:
                j -=1
        return best